packages/bs-ethereum/src/BitqueryEDSEthereum.ts

```diff
@@ -0,0 +1,61 @@
+import { Currency, ExchangeDataService, NetworkType, TokenPricesResponse } from '@cityofzion/blockchain-service'
+import { Client, cacheExchange, fetchExchange, gql } from '@urql/core'
+import fetch from 'node-fetch'
+import { BITQUERY_API_KEY, BITQUERY_URL } from './constants'
+import dayjs from 'dayjs'
+import utc from 'dayjs/plugin/utc'
+import { bitqueryGetPricesQuery } from './graphql'
+
+dayjs.extend(utc)
+export class BitqueryEDSEthereum implements ExchangeDataService {
+  private readonly client: Client
+  private readonly networkType: NetworkType
+
+  constructor(networkType: NetworkType) {
+    this.networkType = networkType
+
+    this.client = new Client({
+      url: BITQUERY_URL,
+      exchanges: [cacheExchange, fetchExchange],
+      fetch,
+      fetchOptions: {
+        headers: {
+          'X-API-KEY': BITQUERY_API_KEY,
+        },
+      },
+    })
+  }
+
+  async getTokenPrices(currency: Currency): Promise<TokenPricesResponse[]> {
+    if (this.networkType !== 'mainnet') throw new Error('Exchange is only available on mainnet')
+
+    const after = dayjs.utc().subtract(2, 'day').startOf('date').toISOString()
```

Comment:
nitpick: after is a confusing name to me, what it really should be is `twoDaysAgo` 👀